From fc459d269159b4a45edbfcad3acf4d5e5253439f Mon Sep 17 00:00:00 2001
From: "Patrick D. Hunt" <phunt@apache.org>
Date: Mon, 10 Feb 2014 21:26:54 +0000
Subject: [PATCH 27/94] ZOOKEEPER-1811. The ZooKeeperSaslClient service name
 principal is hardcoded to "zookeeper" (Harsh J via
 phunt)

git-svn-id: https://svn.apache.org/repos/asf/zookeeper/branches/branch-3.4@1566749 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 6ef87886b5e4ca0e946bd8f35bd347c95c62f914)
---
 src/java/main/org/apache/zookeeper/ClientCnxn.java |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/java/main/org/apache/zookeeper/ClientCnxn.java b/src/java/main/org/apache/zookeeper/ClientCnxn.java
index 27547b4..059c372 100644
--- a/src/java/main/org/apache/zookeeper/ClientCnxn.java
+++ b/src/java/main/org/apache/zookeeper/ClientCnxn.java
@@ -85,6 +85,9 @@ import org.slf4j.LoggerFactory;
 public class ClientCnxn {
     private static final Logger LOG = LoggerFactory.getLogger(ClientCnxn.class);
 
+    private static final String ZK_SASL_CLIENT_USERNAME =
+        "zookeeper.sasl.client.username";
+
     /** This controls whether automatic watch resetting is enabled.
      * Clients automatically reset watches during session reconnect, this
      * option allows the client to turn off this behavior by setting
@@ -940,7 +943,11 @@ public class ClientCnxn {
             setName(getName().replaceAll("\\(.*\\)",
                     "(" + addr.getHostName() + ":" + addr.getPort() + ")"));
             try {
-                zooKeeperSaslClient = new ZooKeeperSaslClient("zookeeper/"+addr.getHostName());
+                String principalUserName = System.getProperty(
+                        ZK_SASL_CLIENT_USERNAME, "zookeeper");
+                zooKeeperSaslClient =
+                    new ZooKeeperSaslClient(
+                            principalUserName+"/"+addr.getHostName());
             } catch (LoginException e) {
                 // An authentication error occurred when the SASL client tried to initialize:
                 // for Kerberos this means that the client failed to authenticate with the KDC.
-- 
1.7.9.5

